play-media-native/features/ImageGrid/ImageOverlayCarousel.tsx

```diff
@@ -0,0 +1,60 @@
+import { Dimensions, Image, StyleSheet } from "react-native";
+import Carousel from "react-native-reanimated-carousel";
+import { IconButton, Modal, Portal } from "react-native-paper";
+
+export const ImageOverlayCarousel = ({
+  images,
+  startIndex = 0,
+  visible,
+  onClose,
+}: {
+  images: string[];
+  startIndex?: number;
+  visible: boolean;
+  onClose: () => void;
+}) => {
+  const width = Dimensions.get("window").width;
+  const height = Dimensions.get("window").height;
+
+  const styles = StyleSheet.create({
+    visibilityStyle: {
+      display: visible ? "flex" : "none",
+    },
+  });
+
+  return (
+    <Portal>
```

Comment:
In my emulator, this throws an error

![image](https://user-images.githubusercontent.com/96472119/214000977-83092f3e-37cb-4c49-a226-eb69f5fe65ad.png)

which can be solved by wrapping everything with Portal.Host for example. Or there might also be another way.